packages/expo-maps/ios/MapPermissionRequester.swift

```diff
@@ -0,0 +1,81 @@
+// Copyright 2025-present 650 Industries. All rights reserved.
+
+import ExpoModulesCore
+import MapKit
+
+class MapPermissionRequester: NSObject, EXPermissionsRequester, CLLocationManagerDelegate {
+  private var locationManager = CLLocationManager()
+  private var resolve: EXPromiseResolveBlock?
+  private var reject: EXPromiseRejectBlock?
+
+  override init() {
+    super.init()
+    locationManager.delegate = self
+  }
+
+  static func permissionType() -> String {
+    return "location"
+  }
+
+  func permissionWith(status systemStatus: CLAuthorizationStatus) -> [AnyHashable: Any] {
+    var status: EXPermissionStatus
+
+    switch systemStatus {
+    case .authorizedAlways, .authorizedWhenInUse:
+      status = EXPermissionStatusGranted
+    case .denied, .restricted:
+      status = EXPermissionStatusDenied
+    case .notDetermined:
+      fallthrough
+    @unknown default:
+      status = EXPermissionStatusUndetermined
+    }
+
+    return [
+      "status": status.rawValue
+    ]
+  }
+
+  func getPermissions() -> [AnyHashable: Any] {
+    var systemStatus: CLAuthorizationStatus
+    let description = Bundle.main.infoDictionary?["NSLocationWhenInUseUsageDescription"] as? String
+
+    if description != nil {
+      systemStatus = locationManager.authorizationStatus
+    } else {
+      EXFatal(EXErrorWithMessage("""
+      This app is missing 'NSLocationWhenInUseUsageDescription',
+      so MapKit services will fail. Add this entry to your bundle's Info.plist.
+      """))
+      systemStatus = .denied
+    }
+
+    return permissionWith(status: systemStatus)
+  }
+
+  func requestPermissions(resolver resolve: @escaping EXPromiseResolveBlock, rejecter reject: @escaping EXPromiseRejectBlock) {
+    if locationManager.authorizationStatus == .notDetermined {
+      locationManager.requestWhenInUseAuthorization()
+      self.resolve = resolve
+      self.reject = reject
+    } else {
+      resolve(getPermissions())
+    }
+  }
+
+  func locationManager(_ manager: CLLocationManager, didChangeAuthorization status: CLAuthorizationStatus) {
+    if resolve != nil {
+      resolve?(getPermissions())
+      resolve = nil
+      reject = nil
+    }
+  }
+
+  func locationManager(_ manager: CLLocationManager, didFailWithError error: any Error) {
+    if reject != nil {
+      reject?("E_LOCATION_MANAGER", "location request failed with error", error)
+      resolve = nil
+      reject = nil
+    }
```

Comment:
```suggestion
    if let reject {
      reject("E_LOCATION_MANAGER", "location request failed with error", error)
      resolve = nil
      self.reject = nil
    }
```
We can't get in these blocks unless it is not nil so I'd prefer to use `if let` and remove the optional chaining